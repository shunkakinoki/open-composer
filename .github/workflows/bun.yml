name: Bun
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
  workflow_dispatch:
concurrency:
  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.head_ref || github.sha }}
  cancel-in-progress: true
jobs:
  bun-build:
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
      - name: Checkout
        uses: actions/checkout@v5
      - name: Setup Bun
        uses: shunkakinoki/actions/.github/actions/setup-bun@ba7baa1682a0248f78110dff5638e9a70b7703b2
      - name: Build
        run: bun run build
  bun-lint:
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
      - name: Checkout
        uses: actions/checkout@v5
      - name: Setup Bun
        uses: shunkakinoki/actions/.github/actions/setup-bun@ba7baa1682a0248f78110dff5638e9a70b7703b2
      - name: Check
        run: bun run check
      - name: Format
        run: bun run format
      - name: Lint
        run: bun run lint
      - name: Type Check
        run: bun run type-check
  bun-test:
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
      - name: Checkout
        uses: actions/checkout@v5
      - name: Setup Bun
        uses: shunkakinoki/actions/.github/actions/setup-bun@ba7baa1682a0248f78110dff5638e9a70b7703b2
      - name: Test
        run: bun run test
  bun-check:
    if: always()
    needs:
      - bun-build
      - bun-lint
      - bun-test
    runs-on: ubuntu-latest
    timeout-minutes: 3
    steps:
      - name: Alls Green
        uses: re-actors/alls-green@release/v1
        with:
          jobs: ${{ toJSON(needs) }}
