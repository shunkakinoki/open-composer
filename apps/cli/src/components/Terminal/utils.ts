/**
 * Terminal utility functions
 * Inspired by termact: https://github.com/MasterGordon/termact/blob/main/src/utils.ts
 */

import type { TerminalStyle } from "./types.js";
import {
  bg,
  blink,
  bold,
  doubleUnderline,
  fg,
  hidden,
  inverse,
  italic,
  strikethrough,
  underline,
} from "./ansi.js";

export const applyStyle = (style: TerminalStyle): string => {
  let result = "";
  if (style.fg) result += fg(style.fg);
  if (style.bg) result += bg(style.bg);
  if (style.bold) result += bold();
  if (style.italic) result += italic();
  if (style.underline) result += underline();
  if (style.blink) result += blink();
  if (style.reverse) result += inverse();
  if (style.hidden) result += hidden();
  if (style.strikethrough) result += strikethrough();
  if (style.doubleUnderline) result += doubleUnderline();
  return result;
};

export const wrapText = (text: string, width: number): string[] => {
  const lines: string[] = [];
  const words = text.split(" ");
  let currentLine = "";

  for (const word of words) {
    if (currentLine.length + word.length + 1 <= width) {
      currentLine += (currentLine ? " " : "") + word;
    } else {
      if (currentLine) lines.push(currentLine);
      currentLine = word;
    }
  }

  if (currentLine) lines.push(currentLine);
  return lines;
};

export const truncateText = (text: string, maxLength: number): string => {
  if (text.length <= maxLength) {
    return text;
  }
  return text.slice(0, maxLength - 3) + "...";
};

export const padText = (
  text: string,
  width: number,
  align: "left" | "center" | "right" = "left",
): string => {
  if (text.length >= width) {
    return text.slice(0, width);
  }

  const padding = width - text.length;

  switch (align) {
    case "center": {
      const leftPad = Math.floor(padding / 2);
      const rightPad = padding - leftPad;
      return " ".repeat(leftPad) + text + " ".repeat(rightPad);
    }
    case "right":
      return " ".repeat(padding) + text;
    default:
      return text + " ".repeat(padding);
  }
};

export const getNtscGrayscale = (r: number, g: number, b: number): number => {
  return r * 0.299 + g * 0.587 + b * 0.114;
};
