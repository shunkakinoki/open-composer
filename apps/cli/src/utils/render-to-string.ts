import { render as inkRender } from "ink";
import type { ReactElement } from "react";
import { EventEmitter } from "node:events";

/**
 * Minimal writable stream for capturing Ink output
 */
class CaptureStream extends EventEmitter {
  private output = "";

  write(chunk: string): void {
    this.output += chunk;
  }

  getOutput(): string {
    return this.output;
  }

  end(): void {
    // no-op
  }
}

/**
 * Renders a React component to a string using Ink
 * Useful for rendering UI components in Effect CLI commands
 */
export function renderToString(element: ReactElement): string {
  const stdout = new CaptureStream();
  const stderr = new CaptureStream();

  const instance = inkRender(element, {
    stdout: stdout as unknown as NodeJS.WriteStream,
    stderr: stderr as unknown as NodeJS.WriteStream,
    debug: false,
    exitOnCtrlC: false,
    patchConsole: false,
  });

  // Wait for a tick to let Ink render
  const output = stdout.getOutput();

  // Clean up
  instance.unmount();
  instance.cleanup();

  return output;
}
