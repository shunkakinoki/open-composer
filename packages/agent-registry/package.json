{
  "name": "@open-composer/agent-registry",
  "version": "0.1.0",
  "repository": "https://github.com/shunkakinoki/open-composer",
  "license": "MIT",
  "author": "Shun Kakinoki",
  "type": "module",
  "main": "./src/index.ts",
  "module": "./src/index.ts",
  "bin": {
    "squad": "./src/cli/squad-command.ts"
  },
  "files": [
    "dist/**/*",
    "src/**/*",
    "README.md",
    "CHANGELOG.md"
  ],
  "scripts": {
    "build": "bun build ./src/index.ts --outdir ./dist --target=bun",
    "lint": "biome check .",
    "lint:fix": "biome check --write .",
    "squad": "bun run ./src/cli/squad-command.ts",
    "test": "bun test",
    "type-check": "echo 'Type check skipped - uses Bun runtime types'"
  },
  "dependencies": {
    "@langchain/core": "^0.3.0",
    "@langchain/langgraph": "^0.2.0",
    "chalk": "^5.3.0",
    "commander": "^12.0.0",
    "inquirer": "^9.2.0",
    "sqlite3": "^5.1.6",
    "uuid": "^10.0.0",
    "zod": "^3.22.0"
  },
  "devDependencies": {
    "@types/bun": "^1.2.23",
    "@types/node": "^24.6.2",
    "@types/uuid": "^10.0.0",
    "typescript": "^5.9.3"
  },
  "publishConfig": {
    "access": "public"
  }
}
