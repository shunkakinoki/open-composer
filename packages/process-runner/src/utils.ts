import * as Effect from "effect/Effect";
import {
  MAX_COMMAND_LENGTH,
  MAX_RUN_NAME_LENGTH,
  RUN_NAME_PATTERN,
} from "./constants.js";
import { ProcessRunnerError } from "./types.js";

// =============================================================================
// VALIDATION UTILITIES
// =============================================================================

export function validateRunName(
  runName: string,
): Effect.Effect<string, ProcessRunnerError> {
  if (!runName || typeof runName !== "string") {
    return Effect.fail(
      ProcessRunnerError("Run name must be a non-empty string"),
    );
  }

  if (runName.length > MAX_RUN_NAME_LENGTH) {
    return Effect.fail(
      ProcessRunnerError(
        `Run name too long (max ${MAX_RUN_NAME_LENGTH} characters)`,
      ),
    );
  }

  if (!RUN_NAME_PATTERN.test(runName)) {
    return Effect.fail(
      ProcessRunnerError(
        "Run name can only contain letters, numbers, hyphens, and underscores",
      ),
    );
  }

  return Effect.succeed(runName);
}

export function validateCommand(
  command: string,
): Effect.Effect<string, ProcessRunnerError> {
  if (!command || typeof command !== "string") {
    return Effect.fail(
      ProcessRunnerError("Command must be a non-empty string"),
    );
  }

  if (command.length > MAX_COMMAND_LENGTH) {
    return Effect.fail(
      ProcessRunnerError(
        `Command too long (max ${MAX_COMMAND_LENGTH} characters)`,
      ),
    );
  }

  // Basic command injection prevention - only reject null bytes
  if (command.includes("\x00")) {
    return Effect.fail(
      ProcessRunnerError("Command contains invalid characters"),
    );
  }

  return Effect.succeed(command);
}

// =============================================================================
// UTILITY FUNCTIONS
// =============================================================================

export function withTimeout<T>(
  effect: Effect.Effect<T, ProcessRunnerError>,
  timeoutMs: number,
  timeoutMessage: string = "Operation timed out",
): Effect.Effect<T, ProcessRunnerError> {
  const timeoutEffect = Effect.async<T, ProcessRunnerError>((resume) => {
    const timeout = setTimeout(() => {
      resume(Effect.fail(ProcessRunnerError(timeoutMessage)));
    }, timeoutMs);
    return Effect.sync(() => clearTimeout(timeout));
  });

  return Effect.race(effect, timeoutEffect);
}
