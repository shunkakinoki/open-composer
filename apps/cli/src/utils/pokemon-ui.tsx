import type { AgentTeam, OpenComposerAgent } from "@open-composer/agent-registry";
import type { RegistryStats } from "@open-composer/analytics";
import { renderToString } from "./render-to-string.js";
import {
  AgentCard,
  AgentComparison,
  AgentList,
  EvolutionTree,
  RegistryStats as RegistryStatsComponent,
  TaskSpecializations,
  TeamDisplay,
} from "../components/PokemonUI.js";

/**
 * Pokemon-style UI renderer using React components
 * This is a utility wrapper that renders React components to strings for CLI output
 */
export class PokemonUI {
  /**
   * Render agent card
   */
  static renderAgentCard(agent: OpenComposerAgent): string {
    return renderToString(<AgentCard agent={agent} />);
  }

  /**
   * Render agent list
   */
  static renderAgentList(agents: OpenComposerAgent[]): string {
    return renderToString(<AgentList agents={agents} />);
  }

  /**
   * Render team display
   */
  static renderTeam(team: AgentTeam): string {
    return renderToString(<TeamDisplay team={team} />);
  }

  /**
   * Render evolution tree
   */
  static renderEvolutionTree(agent: OpenComposerAgent): string {
    return renderToString(<EvolutionTree agent={agent} />);
  }

  /**
   * Render task specializations
   */
  static renderTaskSpecializations(agent: OpenComposerAgent): string {
    return renderToString(<TaskSpecializations agent={agent} />);
  }

  /**
   * Render registry statistics
   */
  static renderStats(stats: RegistryStats): string {
    return renderToString(<RegistryStatsComponent stats={stats} />);
  }

  /**
   * Render agent comparison
   */
  static renderComparison(
    agent1: OpenComposerAgent,
    agent2: OpenComposerAgent,
  ): string {
    return renderToString(<AgentComparison agent1={agent1} agent2={agent2} />);
  }
}
