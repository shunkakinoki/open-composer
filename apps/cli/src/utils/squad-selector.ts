import inquirer from "inquirer";
import type { LLMTask } from "@open-composer/agent-registry";
import { getAgentRegistry, getSquadLauncher } from "@open-composer/agent-registry";

/**
 * Start interactive squad selector
 */
export async function startSquadSelector(): Promise<void> {
  const registry = getAgentRegistry();
  const launcher = getSquadLauncher();

  console.log("\n🎮 Pokemon-Style Squad Manager\n");

  const { action } = await inquirer.prompt([
    {
      type: "list",
      name: "action",
      message: "What would you like to do?",
      choices: [
        { name: "📋 List all agents", value: "list" },
        { name: "✨ Create new squad", value: "create" },
        { name: "🚀 Quick squad (by task)", value: "quick" },
        { name: "📊 View statistics", value: "stats" },
        { name: "🚪 Exit", value: "exit" },
      ],
    },
  ]);

  switch (action) {
    case "list": {
      const agents = registry.getAllAgents();
      console.log(`\n📦 Found ${agents.length} agents:\n`);
      for (const agent of agents) {
        console.log(
          `  ${agent.pokemonAttributes.sprite} ${agent.name} (${agent.tier}) - ${agent.provider}`,
        );
      }
      break;
    }

    case "create": {
      console.log("\n🛠️  Create Squad - Not implemented in CLI yet");
      console.log("   Use the interactive UI components for full squad creation");
      break;
    }

    case "quick": {
      const { task } = await inquirer.prompt([
        {
          type: "list",
          name: "task",
          message: "Select task specialization:",
          choices: [
            { name: "🧭 Planner", value: "planner" },
            { name: "⚡ Programmer", value: "programmer" },
            { name: "🔍 Reviewer", value: "reviewer" },
            { name: "🌊 Router", value: "router" },
            { name: "🌿 Summarizer", value: "summarizer" },
          ],
        },
      ]);

      const team = launcher.createQuickSquad(task as LLMTask);
      console.log(`\n✅ Created squad: ${team.name}`);
      console.log(`   Agents: ${team.agents.map((a) => a.name).join(", ")}`);
      break;
    }

    case "stats": {
      const stats = registry.getStats();
      console.log("\n📊 Registry Statistics:");
      console.log(`   Total Agents: ${stats.totalAgents}`);
      console.log(`   Total Squads: ${stats.totalTeams}`);
      console.log(`   Starter: ${stats.tierCounts.starter || 0}`);
      console.log(`   Evolved: ${stats.tierCounts.evolved || 0}`);
      console.log(`   Legendary: ${stats.tierCounts.legendary || 0}`);
      break;
    }

    case "exit":
      console.log("\n👋 Goodbye!\n");
      return;
  }

  // Loop back to main menu
  await startSquadSelector();
}
