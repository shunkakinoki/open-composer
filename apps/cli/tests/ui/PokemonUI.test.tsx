import { expect, test } from "bun:test";
import {
  AgentCard,
  AgentComparison,
  AgentList,
  EvolutionTree,
  RegistryStats,
  TaskSpecializations,
  TeamDisplay,
} from "../../src/components/PokemonUI.js";
import { render } from "../utils.js";
import type {
  AgentTeam,
  OpenComposerAgent,
} from "@open-composer/agent-registry";
import { LLMTask } from "@open-composer/agent-registry";

// -----------------------------------------------------------------------------
// Mock Data
// -----------------------------------------------------------------------------

const mockAgent: OpenComposerAgent = {
  id: "test-agent-1",
  name: "Claude Sonnet",
  modelName: "claude-sonnet-4",
  provider: "anthropic",
  tier: "evolved",
  taskSpecializations: [LLMTask.PROGRAMMER, LLMTask.REVIEWER],
  pokemonAttributes: {
    sprite: "🤖",
    stats: {
      speed: 85,
      accuracy: 90,
      power: 88,
      efficiency: 92,
      versatility: 87,
    },
    type: "AI/Code",
    evolvesFrom: "Claude Haiku",
    evolvesTo: "Claude Opus",
  },
  compatibility: {
    supportedTasks: [LLMTask.PROGRAMMER, LLMTask.REVIEWER],
    supportsStreaming: true,
    supportsThinking: false,
    maxTokens: 200000,
  },
  description: "Balanced AI agent for coding tasks",
  createdAt: new Date("2024-01-01"),
  performance: {
    totalRequests: 100,
    totalTokensUsed: 50000,
    averageLatency: 1200,
    successRate: 95.0,
  },
};

const mockLegendaryAgent: OpenComposerAgent = {
  ...mockAgent,
  id: "test-agent-2",
  name: "Claude Opus",
  modelName: "claude-opus-4",
  tier: "legendary",
  pokemonAttributes: {
    ...mockAgent.pokemonAttributes,
    sprite: "💎",
    stats: {
      speed: 95,
      accuracy: 98,
      power: 99,
      efficiency: 85,
      versatility: 96,
    },
    evolvesFrom: "Claude Sonnet",
    evolvesTo: undefined,
  },
};

const mockStarterAgent: OpenComposerAgent = {
  ...mockAgent,
  id: "test-agent-3",
  name: "Claude Haiku",
  modelName: "claude-haiku-4",
  tier: "starter",
  pokemonAttributes: {
    ...mockAgent.pokemonAttributes,
    sprite: "⚡",
    stats: {
      speed: 98,
      accuracy: 75,
      power: 65,
      efficiency: 95,
      versatility: 70,
    },
    evolvesFrom: undefined,
    evolvesTo: "Claude Sonnet",
  },
};

const mockTeam: AgentTeam = {
  id: "test-team-1",
  name: "Elite Coding Squad",
  description: "High-performance team for complex coding tasks",
  agents: [mockAgent, mockLegendaryAgent],
  createdAt: new Date("2024-01-01"),
  config: {
    maxConcurrent: 3,
    taskDistribution: "balanced",
    fallbackEnabled: true,
  },
};

const mockStats = {
  totalAgents: 15,
  totalTeams: 5,
  tierCounts: {
    starter: 6,
    evolved: 6,
    legendary: 3,
  },
  providerCounts: {
    anthropic: 9,
    openai: 4,
    "google-genai": 2,
  },
  totalTokensUsed: 1500000,
  totalRequests: 500,
  averageSuccessRate: 94.5,
};

// -----------------------------------------------------------------------------
// AgentCard Tests
// -----------------------------------------------------------------------------

test("AgentCard renders agent details", () => {
  const { lastFrame } = render(<AgentCard agent={mockAgent} />);
  const output = lastFrame();

  expect(output).toContain("Claude Sonnet");
  expect(output).toContain("AI/Code");
  expect(output).toContain("evolved");
  expect(output).toContain("anthropic");
  expect(output).toContain("claude-sonnet-4");
});

test("AgentCard renders stats correctly", () => {
  const { lastFrame } = render(<AgentCard agent={mockAgent} />);
  const output = lastFrame();

  expect(output).toContain("Speed");
  expect(output).toContain("Accuracy");
  expect(output).toContain("Power");
  expect(output).toContain("Efficiency");
  expect(output).toContain("Versatility");
  expect(output).toContain("85");
  expect(output).toContain("90");
  expect(output).toContain("88");
  expect(output).toContain("92");
  expect(output).toContain("87");
});

test("AgentCard renders performance metrics", () => {
  const { lastFrame } = render(<AgentCard agent={mockAgent} />);
  const output = lastFrame();

  expect(output).toContain("Requests");
  expect(output).toContain("100");
  expect(output).toContain("Tokens");
  expect(output).toContain("50000");
  expect(output).toContain("Success");
  expect(output).toContain("95.0");
  expect(output).toContain("Latency");
  expect(output).toContain("1200");
});

test("AgentCard renders evolution information", () => {
  const { lastFrame } = render(<AgentCard agent={mockAgent} />);
  const output = lastFrame();

  expect(output).toContain("Evolution");
  expect(output).toContain("Claude Haiku");
  expect(output).toContain("Claude Opus");
});

test("AgentCard matches snapshot for evolved tier", () => {
  const { lastFrame } = render(<AgentCard agent={mockAgent} />);
  expect(lastFrame()).toMatchSnapshot();
});

test("AgentCard matches snapshot for legendary tier", () => {
  const { lastFrame } = render(<AgentCard agent={mockLegendaryAgent} />);
  expect(lastFrame()).toMatchSnapshot();
});

test("AgentCard matches snapshot for starter tier", () => {
  const { lastFrame } = render(<AgentCard agent={mockStarterAgent} />);
  expect(lastFrame()).toMatchSnapshot();
});

// -----------------------------------------------------------------------------
// AgentList Tests
// -----------------------------------------------------------------------------

test("AgentList renders multiple agents", () => {
  const { lastFrame } = render(
    <AgentList agents={[mockAgent, mockLegendaryAgent, mockStarterAgent]} />,
  );
  const output = lastFrame();

  expect(output).toContain("Claude Sonnet");
  expect(output).toContain("Claude Opus");
  expect(output).toContain("Claude Haiku");
});

test("AgentList renders table headers", () => {
  const { lastFrame } = render(<AgentList agents={[mockAgent]} />);
  const output = lastFrame();

  expect(output).toContain("#");
  expect(output).toContain("Name");
  expect(output).toContain("Type");
  expect(output).toContain("Tier");
  expect(output).toContain("Provider");
});

test("AgentList renders with correct numbering", () => {
  const { lastFrame } = render(
    <AgentList agents={[mockAgent, mockLegendaryAgent]} />,
  );
  const output = lastFrame();

  expect(output).toContain("1");
  expect(output).toContain("2");
});

test("AgentList matches snapshot", () => {
  const { lastFrame } = render(
    <AgentList agents={[mockAgent, mockLegendaryAgent, mockStarterAgent]} />,
  );
  expect(lastFrame()).toMatchSnapshot();
});

test("AgentList matches snapshot with empty list", () => {
  const { lastFrame } = render(<AgentList agents={[]} />);
  expect(lastFrame()).toMatchSnapshot();
});

// -----------------------------------------------------------------------------
// TeamDisplay Tests
// -----------------------------------------------------------------------------

test("TeamDisplay renders team information", () => {
  const { lastFrame } = render(<TeamDisplay team={mockTeam} />);
  const output = lastFrame();

  expect(output).toContain("Elite Coding Squad");
  expect(output).toContain("High-performance team");
});

test("TeamDisplay renders team configuration", () => {
  const { lastFrame } = render(<TeamDisplay team={mockTeam} />);
  const output = lastFrame();

  expect(output).toContain("Configuration");
  expect(output).toContain("Max Concurrent");
  expect(output).toContain("3");
  expect(output).toContain("Task Distribution");
  expect(output).toContain("balanced");
  expect(output).toContain("Fallback");
  expect(output).toContain("true");
});

test("TeamDisplay renders agent members", () => {
  const { lastFrame } = render(<TeamDisplay team={mockTeam} />);
  const output = lastFrame();

  expect(output).toContain("Claude Sonnet");
  expect(output).toContain("Claude Opus");
  expect(output).toContain("evolved");
  expect(output).toContain("legendary");
});

test("TeamDisplay matches snapshot", () => {
  const { lastFrame } = render(<TeamDisplay team={mockTeam} />);
  expect(lastFrame()).toMatchSnapshot();
});

test("TeamDisplay matches snapshot without config", () => {
  const teamWithoutConfig = { ...mockTeam, config: undefined };
  const { lastFrame } = render(<TeamDisplay team={teamWithoutConfig} />);
  expect(lastFrame()).toMatchSnapshot();
});

// -----------------------------------------------------------------------------
// EvolutionTree Tests
// -----------------------------------------------------------------------------

test("EvolutionTree renders evolution chain", () => {
  const { lastFrame } = render(<EvolutionTree agent={mockAgent} />);
  const output = lastFrame();

  expect(output).toContain("Evolution Tree");
  expect(output).toContain("Claude Haiku");
  expect(output).toContain("Claude Sonnet");
  expect(output).toContain("Claude Opus");
});

test("EvolutionTree renders starter agent (no evolvesFrom)", () => {
  const { lastFrame } = render(<EvolutionTree agent={mockStarterAgent} />);
  const output = lastFrame();

  expect(output).toContain("Claude Haiku");
  expect(output).toContain("Claude Sonnet");
  expect(output).not.toContain("Claude Opus");
});

test("EvolutionTree renders legendary agent (no evolvesTo)", () => {
  const { lastFrame } = render(<EvolutionTree agent={mockLegendaryAgent} />);
  const output = lastFrame();

  expect(output).toContain("Claude Sonnet");
  expect(output).toContain("Claude Opus");
});

test("EvolutionTree matches snapshot", () => {
  const { lastFrame } = render(<EvolutionTree agent={mockAgent} />);
  expect(lastFrame()).toMatchSnapshot();
});

// -----------------------------------------------------------------------------
// TaskSpecializations Tests
// -----------------------------------------------------------------------------

test("TaskSpecializations renders task list", () => {
  const { lastFrame } = render(<TaskSpecializations agent={mockAgent} />);
  const output = lastFrame();

  expect(output).toContain("Task Specializations");
  expect(output).toContain("programmer");
  expect(output).toContain("reviewer");
  expect(output).toContain("⚡");
  expect(output).toContain("🔍");
});

test("TaskSpecializations matches snapshot", () => {
  const { lastFrame } = render(<TaskSpecializations agent={mockAgent} />);
  expect(lastFrame()).toMatchSnapshot();
});

// -----------------------------------------------------------------------------
// RegistryStats Tests
// -----------------------------------------------------------------------------

test("RegistryStats renders statistics", () => {
  const { lastFrame } = render(<RegistryStats stats={mockStats} />);
  const output = lastFrame();

  expect(output).toContain("Agent Registry Statistics");
  expect(output).toContain("Total Agents");
  expect(output).toContain("15");
  expect(output).toContain("Total Squads");
  expect(output).toContain("5");
});

test("RegistryStats renders tier counts", () => {
  const { lastFrame } = render(<RegistryStats stats={mockStats} />);
  const output = lastFrame();

  expect(output).toContain("Starter");
  expect(output).toContain("6");
  expect(output).toContain("Evolved");
  expect(output).toContain("Legendary");
  expect(output).toContain("3");
});

test("RegistryStats renders performance metrics", () => {
  const { lastFrame } = render(<RegistryStats stats={mockStats} />);
  const output = lastFrame();

  expect(output).toContain("Performance");
  expect(output).toContain("Total Requests");
  expect(output).toContain("500");
  expect(output).toContain("Total Tokens");
  expect(output).toContain("1500000");
  expect(output).toContain("Avg Success");
  expect(output).toContain("94.5");
});

test("RegistryStats matches snapshot", () => {
  const { lastFrame } = render(<RegistryStats stats={mockStats} />);
  expect(lastFrame()).toMatchSnapshot();
});

// -----------------------------------------------------------------------------
// AgentComparison Tests
// -----------------------------------------------------------------------------

test("AgentComparison renders comparison header", () => {
  const { lastFrame } = render(
    <AgentComparison agent1={mockAgent} agent2={mockLegendaryAgent} />,
  );
  const output = lastFrame();

  expect(output).toContain("Agent Comparison");
  expect(output).toContain("Claude Sonnet");
  expect(output).toContain("Claude Opus");
});

test("AgentComparison renders stat comparisons", () => {
  const { lastFrame } = render(
    <AgentComparison agent1={mockAgent} agent2={mockLegendaryAgent} />,
  );
  const output = lastFrame();

  expect(output).toContain("Speed");
  expect(output).toContain("85");
  expect(output).toContain("95");
  expect(output).toContain("vs");
  expect(output).toContain("Accuracy");
  expect(output).toContain("Power");
  expect(output).toContain("Efficiency");
  expect(output).toContain("Versatility");
});

test("AgentComparison matches snapshot", () => {
  const { lastFrame } = render(
    <AgentComparison agent1={mockAgent} agent2={mockLegendaryAgent} />,
  );
  expect(lastFrame()).toMatchSnapshot();
});

test("AgentComparison matches snapshot with equal stats", () => {
  const { lastFrame } = render(
    <AgentComparison agent1={mockAgent} agent2={mockAgent} />,
  );
  expect(lastFrame()).toMatchSnapshot();
});
