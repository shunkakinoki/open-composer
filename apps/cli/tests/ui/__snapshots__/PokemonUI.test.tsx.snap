// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`AgentCard matches snapshot for evolved tier 1`] = `
"╭──────────────────────────────────────────────────────────────────────────────────────────────────╮
│ 🤖 Claude Sonnet                                                                                 │
│                                                                                                  │
│ Type: AI/Code                                                                                    │
│ Tier: evolved                                                                                    │
│ Provider: anthropic                                                                              │
│ Model: claude-sonnet-4                                                                           │
│                                                                                                  │
│ Stats                                                                                            │
│ Speed        █████████████████░░░  85                                                            │
│ Accuracy     ██████████████████░░  90                                                            │
│ Power        ██████████████████░░  88                                                            │
│ Efficiency   ██████████████████░░  92                                                            │
│ Versatility  █████████████████░░░  87                                                            │
│                                                                                                  │
│ Performance                                                                                      │
│ Requests: 100                                                                                    │
│ Tokens: 50000                                                                                    │
│ Success: 95.0%                                                                                   │
│ Latency: 1200ms                                                                                  │
│                                                                                                  │
│ Evolution:                                                                                       │
│ ← Claude Haiku → Claude Opus                                                                     │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯"
`;

exports[`AgentCard matches snapshot for legendary tier 1`] = `
"╭──────────────────────────────────────────────────────────────────────────────────────────────────╮
│ 💎 Claude Opus                                                                                   │
│                                                                                                  │
│ Type: AI/Code                                                                                    │
│ Tier: legendary                                                                                  │
│ Provider: anthropic                                                                              │
│ Model: claude-opus-4                                                                             │
│                                                                                                  │
│ Stats                                                                                            │
│ Speed        ███████████████████░  95                                                            │
│ Accuracy     ████████████████████  98                                                            │
│ Power        ████████████████████  99                                                            │
│ Efficiency   █████████████████░░░  85                                                            │
│ Versatility  ███████████████████░  96                                                            │
│                                                                                                  │
│ Performance                                                                                      │
│ Requests: 100                                                                                    │
│ Tokens: 50000                                                                                    │
│ Success: 95.0%                                                                                   │
│ Latency: 1200ms                                                                                  │
│                                                                                                  │
│ Evolution:                                                                                       │
│ ← Claude Sonnet                                                                                  │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯"
`;

exports[`AgentCard matches snapshot for starter tier 1`] = `
"╭──────────────────────────────────────────────────────────────────────────────────────────────────╮
│ ⚡ Claude Haiku                                                                                  │
│                                                                                                  │
│ Type: AI/Code                                                                                    │
│ Tier: starter                                                                                    │
│ Provider: anthropic                                                                              │
│ Model: claude-haiku-4                                                                            │
│                                                                                                  │
│ Stats                                                                                            │
│ Speed        ████████████████████  98                                                            │
│ Accuracy     ███████████████░░░░░  75                                                            │
│ Power        █████████████░░░░░░░  65                                                            │
│ Efficiency   ███████████████████░  95                                                            │
│ Versatility  ██████████████░░░░░░  70                                                            │
│                                                                                                  │
│ Performance                                                                                      │
│ Requests: 100                                                                                    │
│ Tokens: 50000                                                                                    │
│ Success: 95.0%                                                                                   │
│ Latency: 1200ms                                                                                  │
│                                                                                                  │
│ Evolution:                                                                                       │
│ → Claude Sonnet                                                                                  │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯"
`;

exports[`AgentList matches snapshot 1`] = `
"#   Name                     Type                Tier           Provider
───────────────────────────────────────────────────────────────────────────────
1   🤖 Claude Sonnet          AI/Code             evolved        anthropic
2   💎 Claude Opus            AI/Code             legendary      anthropic
3   ⚡ Claude Haiku           AI/Code             starter        anthropic"
`;

exports[`AgentList matches snapshot with empty list 1`] = `
"#   Name                     Type                Tier           Provider
───────────────────────────────────────────────────────────────────────────────"
`;

exports[`TeamDisplay matches snapshot 1`] = `
"╭──────────────────────────────────────────────────────────────────────────────────────────────────╮
│ SQUAD: Elite Coding Squad                                                                        │
│ High-performance team for complex coding tasks                                                   │
│                                                                                                  │
│ Configuration:                                                                                   │
│   Max Concurrent: 3                                                                              │
│   Task Distribution: balanced                                                                    │
│   Fallback: true                                                                                 │
│                                                                                                  │
│ 1. 🤖 Claude Sonnet                                                                              │
│    evolved | anthropic | claude-sonnet-4                                                         │
│ 2. 💎 Claude Opus                                                                                │
│    legendary | anthropic | claude-opus-4                                                         │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯"
`;

exports[`TeamDisplay matches snapshot without config 1`] = `
"╭──────────────────────────────────────────────────────────────────────────────────────────────────╮
│ SQUAD: Elite Coding Squad                                                                        │
│ High-performance team for complex coding tasks                                                   │
│                                                                                                  │
│ 1. 🤖 Claude Sonnet                                                                              │
│    evolved | anthropic | claude-sonnet-4                                                         │
│ 2. 💎 Claude Opus                                                                                │
│    legendary | anthropic | claude-opus-4                                                         │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯"
`;

exports[`EvolutionTree matches snapshot 1`] = `
"Evolution Tree for Claude Sonnet:

  Claude Haiku
    ↓
  Claude Sonnet 🤖
    ↓
  Claude Opus"
`;

exports[`TaskSpecializations matches snapshot 1`] = `
"Task Specializations for Claude Sonnet:

  ⚡ programmer
  🔍 reviewer"
`;

exports[`RegistryStats matches snapshot 1`] = `
"╭──────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Agent Registry Statistics                                                                        │
│                                                                                                  │
│ Total Agents: 15                                                                                 │
│ Total Squads: 5                                                                                  │
│                                                                                                  │
│ Tiers                                                                                            │
│   Starter: 6                                                                                     │
│   Evolved: 6                                                                                     │
│   Legendary: 3                                                                                   │
│                                                                                                  │
│ Performance                                                                                      │
│ Total Requests: 500                                                                              │
│ Total Tokens: 1500000                                                                            │
│ Avg Success: 94.5%                                                                               │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯"
`;

exports[`AgentComparison matches snapshot 1`] = `
"Agent Comparison: Claude Sonnet vs Claude Opus

  Speed            85 vs  95
  Accuracy         90 vs  98
  Power            88 vs  99
  Efficiency       92 vs  85
  Versatility      87 vs  96"
`;

exports[`AgentComparison matches snapshot with equal stats 1`] = `
"Agent Comparison: Claude Sonnet vs Claude Sonnet

  Speed            85 vs  85
  Accuracy         90 vs  90
  Power            88 vs  88
  Efficiency       92 vs  92
  Versatility      87 vs  87"
`;
